File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    visited = [False] * n
    answer = 0
    
    def dfs(node):
        visited[node] = True
        for i in range(n):
            if computers[node][i] == 1 and not visited[i]:
                dfs(i)
                
    for i in range(n):
        if not visited[i]:
            dfs(i)
            answer +=1
    
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.cs
using System;

public class Solution {
    public int solution(int n, int[,] computers) {
        int cnt = 0;
        bool[] visited = new bool[n];
        
        for(int i=0; i<n; i++){
            if(!visited[i]){
                DFS(i, n, computers, visited);
                cnt++;
            }
            
        }
        
        return cnt;
    }
    void DFS(int node, int n, int[,] computers, bool[] visited){
        visited[node] = true;
        for(int i = 0; i<n; i++){
            if(!visited[i] && computers[node, i] == 1){
                DFS(i, n, computers, visited);
        }
    }
}
}